pkg/filters/proxy/loadbalance.go

```diff
@@ -38,17 +40,21 @@ const (
 	LoadBalancePolicyIPHash = "ipHash"
 	// LoadBalancePolicyHeaderHash is the load balance policy of HTTP header hash.
 	LoadBalancePolicyHeaderHash = "headerHash"
+	// LoadBalancePolicyStickySessions is the load balance policy of sticky session.
+	LoadBalancePolicyStickySessions = "stickySessions"
 )
 
 // LoadBalancer is the interface of an HTTP load balancer.
 type LoadBalancer interface {
-	ChooseServer(req *httpprot.Request) *Server
+	ChooseServer(spCtx *serverPoolContext) *Server
```

Comment:
As we are discussing in Discord, there's a plan to make the LoadBalancer more generic, that's independent to protocols, change the argument type to `serverPoolContext` makes things harder.